w08h03/test/pgdp/datastructures/UnitTest.java

```diff
@@ -36,6 +42,90 @@ <Type extends Comparable<Type>> void testArray(Type[] input) {
 				"Invalid Iteration Count. Expected [" + expected.length + "] got [" + position + "]");
 	}
 
+	@Test
+	@DisplayName("Read 1M nodes")
+	public void testPerformance1M() {
+		int amountOfValues = (int) 1e6;
+
+		List<Integer> values = IntStream.range(0, amountOfValues).boxed().collect(Collectors.toList());
+		List<Integer> solution = IntStream.range(0, amountOfValues).boxed().collect(Collectors.toList());
+		Collections.shuffle(values);
+
+		QuarternarySearchTree<Integer> n = new QuarternarySearchTree<Integer>();
+		for (Integer i : values) {
+			n.insert(i);
+		}
+
+		ArrayList<Integer> actual = new ArrayList<>();
+
+		System.out.println("Keep in mind the RAM usage and its differences during iteration is not a terribly useful metric here:");
+		long memUsageMBBefore = (Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory())
+				/ (1024 * 1024);
+		System.out.println("Memory usage before iteration: " + memUsageMBBefore + "MB");
+
+		long startTime = System.nanoTime();
+
+		for (Integer r : n) {
+
+			if (r.intValue() == 0) {
+				long memUsageMB = (Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory())
+						/ (1024 * 1024);
+				System.out.println("Memory usage at the start of iteration: " + memUsageMB + "MB");
+			} else if (r.intValue() == amountOfValues - 1) {
```

Comment:
das intValue() könntest du dir prinzipiell sparen, wenn du hier statt Integer int nutzt.
Null-Safe wäre der Loop so oder so nicht, da du durch r.intValue() explizit davon ausgehst, dass r != null ist.

```suggestion
		for (int r : n) {

			if (r == 0) {
				long memUsageMB = (Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory())
						/ (1024 * 1024);
				System.out.println("Memory usage at the start of iteration: " + memUsageMB + "MB");
			} else if (r == amountOfValues - 1) {
```

Reply:
Stimmt, danke